### ISSUE TITLE
Padding Discrepancies On Front End Vs Visual Builder

### ISSUE THREAD

**Issue by timayhelen** [elegantthemes/Divi-Beta#450]
I notice that sometimes there is padding that exists in the Visual Builder that does not exist on the website front end. It was due to padding values for sections & rows being input as "0" without any units via the backend builder. In the Visual Builder, these are interpreted exactly as "0" whereas in on the front-end they are output as "0px." 

In the visual builder, we do not allow people to input "0" because it auto-corrects to 0px. This is not the case in the BB, however. 

So, basically, we need to add "px" to padding values if they have no units defined in the Visual Builder output. We should be sure not to use this method for things that do not require units, such as Gutter Width. I think this only applies to padding, since we use a different input style.

**Comment by Jronk** [comment]
0 is the same as 0px though, and style best practices say you need to omit the px in 0px, in fact: `0 = 0px = 1pc = 0em` etc...

**Comment by Jronk** [comment]
![screen shot 2016-09-12 at 10 17 26 pm](https://cloud.githubusercontent.com/assets/1163441/18462403/c6db7e10-7936-11e6-855f-4dec49b6ef79.png)

**Comment by timayhelen** [comment]
@Jronk I must be wrong about what's actually happening then. But the root cause is the same, so we need to figure out why entering "0" works on the front end, but not in the visual builder.

**Comment by SlavaET** [comment]
@timayhelen I think I found the root cause. If I understand the issue correctly, you're talking about the difference in the bottom spacing of Sections, it's caused by this code:
https://github.com/elegantthemes/submodule-builder/commit/4347008c
So we add the `:after` element which takes some space at the bottom of column:
![image](https://cloud.githubusercontent.com/assets/3446682/18470782/25ef4f8c-79d0-11e6-938a-6c04fd023598.png)

I'm not sure that understand what this code supposed to fix. @fikrirasyid do you remember which issue you fixed with that code? Looks like we need to think of a different solution which will not affect the height of columns in Visual Builder.

**Comment by fikrirasyid** [comment]
@SlavaET it was added to prevent collapsing column. If a column has no module, we have customized it to avoid collapsing column. However if a column has only one module and somehow that module is failed to be loaded, the column collapses. The `:after` styling was added to avoid that kind of collapsing behaviour.

**Comment by timayhelen** [comment]
@SlavaET That's actually different than the bug I am experiencing. I guess in my case, the "0" padding values are not being applied at all. They work on the front end, but not in the visual builder. 

Front End:

![screen shot 2016-09-13 at 10 38 39 am](https://cloud.githubusercontent.com/assets/3333028/18484673/447bb45e-799e-11e6-97af-0159355a0157.png)

Visual Builder:

![screen shot 2016-09-13 at 10 39 04 am](https://cloud.githubusercontent.com/assets/3333028/18484687/528fbafe-799e-11e6-8f84-8617a66a9e0a.png)

Padding shows this in the visual builder:

![screen shot 2016-09-13 at 10 39 46 am](https://cloud.githubusercontent.com/assets/3333028/18484717/72bb31e6-799e-11e6-99f3-f19b66ba4f8a.png)

But it shows this on the front end:

![screen shot 2016-09-13 at 10 40 29 am](https://cloud.githubusercontent.com/assets/3333028/18484738/886af422-799e-11e6-9eff-b0985d4a24ee.png)

So in the visual builder, the bottom padding is being applied to padding-right instead of padding-bottom. It could be a legacy options issue, because once I update it in the Visual Builder it fixes it.

**Comment by andyhqtran** [comment]
The reason the spacing is visible is because it's applying an invisible div to the end of the block so even if we set it to `padding: 0;` it should still be there. Until we set it to `display: none`, it'll still be there.

**Comment by fikrirasyid** [comment]
@SlavaET the issue you mentioned at https://github.com/elegantthemes/Divi-Beta/issues/450#issuecomment-246642627 has been reported at https://github.com/elegantthemes/Divi-Beta/issues/451 and a PR for it has been submitted at https://github.com/elegantthemes/submodule-builder/pull/1173
